### ISSUE TITLE
--vm-driver=none stops all docker containers

### ISSUE THREAD

**Issue by hutchic** [kubernetes/minikube#4664]
A side effect of https://github.com/kubernetes/minikube/commit/df5bbc3c8ae54745ba804eeeb2792ec7e70fe516 is now minikube commands will stop containers that are unrelated to running minikube

**Comment by medyagh** [comment]
tagging @afbjorklund  as he knows this better

**Comment by afbjorklund** [comment]
The code should still use the same "kubernetes prefix" (`k8s_`) for Docker, the change was only for the other runtimes.

When running with a runtime such as cri-o, we will use the standard `crictl` command to _only_ list Kubernetes containers.
The ones running with e.g. podman are in a separate listing anyway, it is only the images that are shared between them.

**Comment by hutchic** [comment]
I believe you're correct. I'm not sure what's changed but for whenever I do `minikube delete` it nukes all my running docker containers (looks like it's restarting the docker daemon)

```
docker version 
Client: Docker Engine - Community
 Version:           19.03.0-beta4
 API version:       1.40
 Go version:        go1.12.4
 Git commit:        e4666eb
 Built:             Tue May 14 12:51:53 2019
 OS/Arch:           linux/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.0-beta4
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.12.4
  Git commit:       e4666eb
  Built:            Tue May 14 12:50:26 2019
  OS/Arch:          linux/amd64
  Experimental:     true
 containerd:
  Version:          1.2.4
  GitCommit:        e6b3f5632f50dbc4e9cb6288d911bf4f5e95b18e
 runc:
  Version:          1.0.0-rc6+dev
  GitCommit:        6635b4f0c6af3810594d2770f662f34ddc15b40d
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
```

```
uname -a
Linux ip-172-31-24-121 4.4.0-1087-aws #98-Ubuntu SMP Wed Jun 26 05:50:53 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
```

```
lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.6 LTS
Release:	16.04
Codename:	xenial
```

journalctl immediately after doing `minikube delete`
```
Jul 03 17:49:04 ip-172-31-24-121 kernel: docker0: port 3(veth4393861) entered forwarding state
Jul 03 17:49:05 ip-172-31-24-121 kernel: docker0: port 2(veth7371c3b) entered forwarding state
Jul 03 17:49:07 ip-172-31-24-121 kernel: docker0: port 4(veth443f232) entered forwarding state
Jul 03 17:49:08 ip-172-31-24-121 kernel: docker0: port 6(vetha4368f7) entered forwarding state
Jul 03 17:49:08 ip-172-31-24-121 kernel: docker0: port 5(vethb243ec0) entered forwarding state
Jul 03 17:49:08 ip-172-31-24-121 kernel: docker0: port 7(veth3025311) entered forwarding state
Jul 03 17:49:10 ip-172-31-24-121 sudo[15299]:   ubuntu : TTY=pts/0 ; PWD=/home/ubuntu ; USER=root ; COMMAND=/bin/bash -c sudo systemctl stop kubelet.service
                                                      sudo rm -rf /data/minikube
                                                      sudo rm -rf /etc/kubernetes/manifests
                                                      sudo rm -rf /var/lib/minikube || true
```

**Comment by hutchic** [comment]
this doesn't happen if I run `minikube delete` without `sudo`

I'm comfortable with that as a solution although it still feels like something changed :thinking:

**Comment by palnabarun** [comment]
@afbjorklund I think this commit https://github.com/kubernetes/minikube/commit/ec135dc4dabfa6ef88cce51ec25cc80ba4123587#diff-3de31142ccf24256d20634e11699a0d4 reverted back some of the changes that were committed in https://github.com/kubernetes/minikube/commit/df5bbc3c8ae54745ba804eeeb2792ec7e70fe516 . 

I added those back locally and tested the functionality. The `minikube` that I built with those changes stopped deleting containers created by me, it only killed those containers created by minikube.

Shall I file a PR with those changes?

Specifically,

```
diff --git a/pkg/minikube/cruntime/docker.go b/pkg/minikube/cruntime/docker.go
index a113668eb..a42e1d106 100644
--- a/pkg/minikube/cruntime/docker.go
+++ b/pkg/minikube/cruntime/docker.go
@@ -25,6 +25,9 @@ import (
        "k8s.io/minikube/pkg/minikube/console"
 )
 
+// KubernetesContainerPrefix is the prefix of each kubernetes container
+const KubernetesContainerPrefix = "k8s_"
+
 // Docker contains Docker runtime state
 type Docker struct {
        Socket string
@@ -102,6 +105,7 @@ func (r *Docker) KubeletOptions() map[string]string {
 
 // ListContainers returns a list of containers
 func (r *Docker) ListContainers(filter string) ([]string, error) {
+       filter = KubernetesContainerPrefix + filter
        content, err := r.Runner.CombinedOutput(fmt.Sprintf(`docker ps -a --filter="name=%s" --format="{{.ID}}"`, filter))
        if err != nil {
                return nil, err
```

**Comment by afbjorklund** [comment]
@palnabarun : yeah, that was unfortunate - we need the extra filter if we run docker (rather than crictl)

**Comment by afbjorklund** [comment]
Should be fixed now ?

**Comment by palnabarun** [comment]
@afbjorklund Yes. It should be fixed with #4689

**Comment by tstromberg** [comment]
Fixed at head, but I will leave this open until it's included in a release.
